### PR TITLE
Basic implementation of survey sections

### PR DESCRIPTION
**Pull request by kxmbrian**
After discussion with @allenwq, decided to sacrifice flexibility of layout (e.g. allow surveys to contain questions directly without sections) and enforce that questions belong to sections. While the nesting adds some complexity, the overall UI is simpler and easier to manage.

Next step of #1020

### CODE REVIEW

File: app/controllers/course/survey/surveys_controller.rb [hunk 1]
```diff
@@ -6,7 +6,9 @@ class Course::Survey::SurveysController < Course::ComponentController
   def index
     respond_to do |format|
       format.html
-      format.json
+      format.json do
+        @surveys = @surveys.includes(:responses)
```
Comment (allenwq) [inline app/controllers/course/survey/surveys_controller.rb]: It might be even slower when there are lots of students in the course. We should only load the responses of the current student. 

Do you want to make a separate PR for this ?

Comment (kxmbrian) [inline app/controllers/course/survey/surveys_controller.rb]: Sure, removed from this PR!

File: app/views/course/survey/sections/_section.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,6 @@
+json.(section, :id, :title, :description, :weight)
+questions = @questions || section.questions
+json.questions questions, partial: 'course/survey/questions/question', as: :question
+json.canCreateQuestion can?(:create, section.questions.build)
```
Comment (allenwq) [inline app/views/course/survey/sections/_section.json.jbuilder]: `section.questions.build` will add a new empty question to the list of questions, which is unexpected ?  I think it's better to `Question.new(section:)` instead.

File: client/app/api/course/Survey/Responses.js [hunk 3]
```diff
@@ -2,19 +2,27 @@ import BaseSurveyAPI from './Base';
...
-  *        - Array of questions belonging to the survey
-  *        - question_type is one of ['text', 'multiple_choice', 'multiple_response']
   *   },
   *   response: {
   *     id:number, submitted_at:datetime,
   *       - Response Attributes
-  *     answers: Array.<question_id:number, text_response:string, options:Array. etc...>,
-  *       - Answer attributes
+  *     sections:
+  *       Array.<
```
Comment (allenwq) [inline client/app/api/course/Survey/Responses.js]:
```js
Array.<{
   id: number, 
}>
```

Missing `{` after `Array`, `{}` indicates it's an object. And let's add a space before the types (number/string)? , I am not sure why some of my comments also missing the spaces, will add them later.

File: app/views/course/survey/sections/_section.json.jbuilder (hunk 2 above)
Comment (kxmbrian) [inline app/views/course/survey/sections/_section.json.jbuilder]: Fixed, thanks!

File: client/app/api/course/Survey/Responses.js (hunk 3 above)
Comment (kxmbrian) [inline client/app/api/course/Survey/Responses.js]: Fixed, thanks!

File: app/views/course/survey/surveys/_survey.json.jbuilder [hunk 4]
```diff
@@ -4,7 +4,7 @@ json.(survey, :id, :title, :description, :start_at, :end_at, :base_exp, :publish
 json.isStarted current_user_response.present?
 json.responseId current_user_response && current_user_response.id
 json.submitted_at current_user_response && current_user_response.submitted_at
-json.canCreateQuestion can?(:create, survey.questions.build)
 json.canUpdate can?(:update, survey)
 json.canDelete can?(:destroy, survey)
+json.canCreateSection can?(:create, survey.sections.build)
```
Comment (allenwq) [inline app/views/course/survey/surveys/_survey.json.jbuilder]: @kxmbrian There's still some occurrences

File: client/app/api/course/Survey/Responses.js (hunk 3 above)
Comment (allenwq) [inline client/app/api/course/Survey/Responses.js]: thanks !!

**Comment by allenwq** [comment]
LGTM.

😛 @kxmbrian Going to work on tests for surveys next ?

**Comment by kxmbrian** [comment]
Yeah, looking at your assessment tests now!
